diff -r 0b576ab589c5 Python/random.c
--- a/Python/random.c	Sat Nov 12 14:37:11 2016 +0200
+++ b/Python/random.c	Sat Nov 12 19:29:25 2016 +0000
@@ -12,9 +12,10 @@
 #  ifdef HAVE_LINUX_RANDOM_H
 #    include <linux/random.h>
 #  endif
-#  ifdef HAVE_GETRANDOM
+#  if defined(HAVE_GETRANDOM) || defined(HAVE_GETENTROPY)
 #    include <sys/random.h>
-#  elif defined(HAVE_GETRANDOM_SYSCALL)
+#  endif
+#  if !defined(HAVE_GETRANDOM) && defined(HAVE_GETRANDOM_SYSCALL)
 #    include <sys/syscall.h>
 #  endif
 #endif
